lib/logstash/inputs/syslog.rb

```diff
@@ -133,39 +153,46 @@ def udp_listener(output_queue)
 
   private
   def tcp_listener(output_queue)
+    @tcp_sockets = []
     @logger.info("Starting syslog tcp listener", :address => "#{@host}:#{@port}")
     @tcp = TCPServer.new(@host, @port)
-    @tcp_clients = []
```

Comment:
OMG this is nasty - `@tcp_clients` / `@tcp_sockets` is reinitialized as a non thread-safe Array here - it was previously initialized as a `ThreadSafe::Array` in `register`.